Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedListItemView.swift

```diff
@@ -0,0 +1,133 @@
+    @ObservedObject var preplannedMapModel: PreplannedMapModel
+    
+    /// A Boolean value indicating whether the preplanned map area can be downloaded. This is `true`
+    /// when the preplanned map area is loaded and the packaging status is `.complete`, and `false`
+    /// when the packaging status is `.processing`. If the preplanned map area has not yet loaded then
+    /// this is `nil`,
+    @State private var canDownload: Bool?
+    
+    /// The error for the preplanned map.
+    @State var error: Error?
+    
+    public var body: some View {
+        HStack {
+            HStack {
+                let preplannedMapArea = preplannedMapModel.preplannedMapArea
+                if let thumbnail = preplannedMapModel.preplannedMapArea.portalItem.thumbnail {
+                    LoadableImageView(loadableImage: thumbnail)
+                        .frame(width: 64, height: 44)
+                }
+                VStack(alignment: .leading, spacing: 2) {
+                    Text(preplannedMapArea.portalItem.title)
+                        .font(.headline)
+                    if !preplannedMapArea.portalItem.description.isEmpty {
+                        Text(preplannedMapArea.portalItem.description)
+                            .font(.caption)
+                            .foregroundStyle(Color(uiColor: .secondaryLabel))
+                            .lineLimit(2)
+                    } else {
+                        Text("The area has no description.")
+                            .font(.caption)
+                            .foregroundStyle(Color(uiColor: .tertiaryLabel))
+                    }
+                    if let error {
+                        Text(error.localizedDescription)
+                            .foregroundStyle(.red)
+                    }
+                }
+                Spacer()
+                switch preplannedMapModel.result {
+                case .success:
+                    Image(systemName: "checkmark.circle.fill")
+                case .failure(let error):
+                    Image(systemName: "exclamationmark.circle")
+                        .foregroundStyle(.red)
+                        .onAppear {
+                            self.error = error
+                        }
+                case .none:
+                    if let canDownload {
+                        if !canDownload {
+                            // Map is still packaging.
+                            VStack {
+                                Image(systemName: "clock.badge.xmark")
+                                Text("packaging")
+                                    .font(.footnote)
+                            }
+                            .foregroundStyle(.secondary)
+                        } else {
+                            // Map package is available for download.
+                            Image(systemName: "arrow.down.circle")
+                                .foregroundStyle(Color.accentColor)
```

Comment:
you don't need to manually set the color. you just need to use a button with an image as the content of the button.